File: 351003/Pisarchik/notebook/src/main/java/com/lizaveta/notebook/mapper/NoticeMapper.java
package com.lizaveta.notebook.mapper;

import com.lizaveta.notebook.model.dto.request.NoticeRequestTo;
import com.lizaveta.notebook.model.dto.response.NoticeResponseTo;
import com.lizaveta.notebook.model.entity.Notice;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingConstants;
import org.mapstruct.ReportingPolicy;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface NoticeMapper {

    @Mapping(target = "id", ignore = true)
    Notice toEntity(NoticeRequestTo request);

    NoticeResponseTo toResponse(Notice entity);
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/mapper/NoticeMapper.kt
package com.distcomp.mapper

import com.distcomp.dto.notice.NoticeRequestTo
import com.distcomp.dto.notice.NoticeResponseTo
import com.distcomp.entity.Notice
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingConstants
import org.mapstruct.ReportingPolicy

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface NoticeMapper {
    @Mapping(source = "news.id", target = "newsId")
    fun toNoticeResponse(notice: Notice) : NoticeResponseTo

    fun toNoticeEntity(noticeRequestTo: NoticeRequestTo) : Notice
}